### PR TITLE
Update data view when posts no longer match search/filters

### PR DESCRIPTION
**Pull request by crcommons**
This pull request makes the following changes:
- Checks to see if, after a post has been edited, the post still matches existing filters;
- If it does not, it removes the post from the post list
- if it does still match, it does nothing

Testing checklist:
One example (can be replaced with any search)
- [ ] POST IS REMOVED FROM LIST WHEN NO LONGER MATCHING FILTERS
  - [ ] Filter posts to only show published posts
  - [ ] Select 'edit' on any post
  - [ ] Change status from Published to Under Review; select Save
  - [ ] Post should briefly change, then disappear from data view and detail view; next card in line should be select (unless you edited the last post in the list, then the selected post will be null)

- [ ] POST REMAINS IN LIST WHEN IT STILL MATCHES FILTERS 
  - [ ] Filter posts to only show published posts
  - [ ] Select 'edit' on any post
  - [ ] Make a change in the post, but leave the status set to Published; select Save
  - [ ] Postcard and post detail should update with change, but should remain in view

- [ ] Try this again using any and all filters!

Fixes ushahidi/platform#2227

Ping @ushahidi/platform

### CODE REVIEW

File: app/main/posts/views/post-view-data.directive.js [hunk 1]
```diff
@@ -88,6 +88,41 @@ function PostViewDataController(
...
+            let query = PostFilters.getQueryParams($scope.filters);
+
+            let postQuery = _.extend({}, query, {
+                limit: $scope.itemsPerPage
+            });
+
+            postQuery.offset = ($scope.currentPage - 1) * $scope.itemsPerPage;
+
+            PostEndpoint.query(postQuery).$promise.then(function (postsResponse) {
+                for (let i = 0; i < postsResponse.results.length; i++) {
```
Comment (rjmackay) [inline app/main/posts/views/post-view-data.directive.js]: Why not just use foreach?

File: app/main/posts/views/post-view-data.directive.js [hunk 2]
```diff
@@ -88,6 +88,41 @@ function PostViewDataController(
...
+                    if (postsResponse.results[i].id === $scope.selectedPostId) {
+                        return;
+                    }
+                }
+                removePostFromList();
```
Comment (rjmackay) [inline app/main/posts/views/post-view-data.directive.js]: I don't understand what this is doing? it looks like you're *always* removing it from the list?

File: app/main/posts/views/post-view-data.directive.js [hunk 3]
```diff
@@ -88,6 +88,41 @@ function PostViewDataController(
...
+
+            let query = PostFilters.getQueryParams($scope.filters);
+
+            let postQuery = _.extend({}, query, {
+                limit: $scope.itemsPerPage
+            });
+
+            postQuery.offset = ($scope.currentPage - 1) * $scope.itemsPerPage;
+
+            PostEndpoint.query(postQuery).$promise.then(function (postsResponse) {
```
Comment (rjmackay) [inline app/main/posts/views/post-view-data.directive.js]: ... I'm missing something. Why are we reloading?

File: app/common/services/endpoints/post-endpoint.js [hunk 4]
```diff
@@ -22,58 +22,24 @@ function (
...
-                    if (!data) {
-                        return data;
-                    }
-
-                    data.results = data.results.map(normalizePost);
-
-                    return data;
-                }
-            ]
+            }
```
Comment (rjmackay) [inline app/common/services/endpoints/post-endpoint.js]: Accidentally reverted changes??

File: app/main/posts/views/post-view-data.directive.js [hunk 5]
```diff
@@ -88,6 +88,36 @@ function PostViewDataController(
      */
     var newPostsAfter = moment().utc();
     $scope.savingPost = {saving: false};
+
+    $rootScope.$on('event:edit:post:data:mode:saveSuccess', function () {
+        function removePostFromList() {
+            for (var i = 0; i < $scope.posts.length; i++) {
```
Comment (rjmackay) [inline app/main/posts/views/post-view-data.directive.js]: I'd generally suggest just using underscore for this ie. 
```
let index = _.findIndex($scope.posts, { id: selectedId });
$scope.posts.splice(i, 1);
```
Unless we see a real slowdown from doing this, I favour readability

File: app/common/services/endpoints/post-endpoint.js (hunk 4 above)
Comment (crcommons) [reply app/common/services/endpoints/post-endpoint.js]: that was on purpose. When I wasn't getting the correct thing back from the DB, I made some unnecessary changes.

**Pull request merged**